backend/src/routes.rs

```diff
@@ -0,0 +1,154 @@
+    pagination_query: web::Query<Query>,
+) -> Result<HttpResponse, AppError> {
+    let pagination = pagination_query.into_inner().into();
+
+    let content_range = get_attestations_count(&state.db_executor).await;
+
+    let attestation_requests = get_attestation_requests(pagination, &state.db_executor).await?;
+
+    let response = serde_json::json!(attestation_requests);
+
+    Ok(HttpResponse::Ok()
+        .insert_header(("Content-Range", content_range))
+        .json(response))
+}
+
+#[delete("/{attestation_request_id}")]
+async fn delete_attestation(
+    path: web::Path<Uuid>,
+    state: web::Data<AppState>,
+) -> Result<HttpResponse, AppError> {
+    let attestation_id = path.into_inner();
+    delete_attestation_request(&attestation_id, &state.db_executor).await?;
+    log::info!("Attestation with id {:?} is deleted", attestation_id);
+    Ok(HttpResponse::Ok().json("ok"))
+}
+
+#[post("")]
+async fn post_attestation(
+    body: web::Json<AttestationRequest>,
+    state: web::Data<AppState>,
+) -> Result<HttpResponse, AppError> {
+    let attestation_request = body.into_inner();
+    let credential: Credential = serde_json::from_value(attestation_request.clone().credential)?;
+    let attestation =
+        insert_attestation_request(&attestation_request, &credential, &state.db_executor).await?;
+    log::info!(" New attestation with id {:?} is created", attestation.id);
+    Ok(HttpResponse::Ok().json(attestation))
+}
+
+#[put("/{attestation_request_id}/approve")]
+async fn approve_attestation(
+    path: web::Path<Uuid>,
+    state: web::Data<AppState>,
+) -> Result<HttpResponse, AppError> {
+    let attestation_id = path.into_inner();
+    let attestation = can_approve_attestation(&attestation_id, &state.db_executor).await?;
+    let credential = serde_json::from_value::<Credential>(attestation.credential)?;
+    let ctype_hash = hex::decode(credential.claim.ctype_hash.trim_start_matches("0x").trim())?;
+    let claim_hash = hex::decode(credential.root_hash.trim_start_matches("0x").trim())?;
+    if claim_hash.len() != 32 || ctype_hash.len() != 32 {
+        return Ok(HttpResponse::BadRequest().json("Claim hash or ctype hash have a wrong format"));
+    }
+
+    crate::tx::create_claim(
+        H256::from_slice(&claim_hash),
+        H256::from_slice(&ctype_hash),
+        state.config.clone(),
+    )
+    .await?;
+
+    approve_attestation_request(&attestation_id, &state.db_executor).await?;
+    log::info!("Attestation with id {:?} is approved", attestation_id);
+    Ok(HttpResponse::Ok().json("ok"))
+}
+
+#[put("/{attestation_request_id}/revoke")]
+async fn revoke_attestation(
+    path: web::Path<Uuid>,
+    state: web::Data<AppState>,
+) -> Result<HttpResponse, AppError> {
+    let attestation_id = path.into_inner();
+
+    let attestation = can_revoke_attestation(&attestation_id, &state.db_executor).await?;
+
+    let credential = serde_json::from_value::<Credential>(attestation.credential)?;
```

Comment:
```suggestion
    let credential: Credential = serde_json::from_value(attestation.credential)?;
```

Reply:
Made it consist in the project: [a0d7b04](https://github.com/KILTprotocol/attester/pull/1/commits/a0d7b042cf735da6244f8c390298eee8d4195dae)